### ISSUE TITLE
Core functionality of `/data_objects/study/{study_id}` endpoint is not being tested

### ISSUE THREAD

**Issue by eecavanna** [microbiomedata/nmdc-runtime#725]
The only test that targets the `/data_objects/study/{study_id}` endpoint, targets a corner case where the specified study does not exist.

https://github.com/microbiomedata/nmdc-runtime/blob/bcdf8f390974786f879806123b28f926881b056c/tests/test_api/test_endpoints.py#L349-L364

There are no tests that target the core functionality (a.k.a. "happy path") of the endpoint.

https://github.com/microbiomedata/nmdc-runtime/blob/bcdf8f390974786f879806123b28f926881b056c/tests/test_api/test_endpoints.py#L36-L38

Previously, there was a test that targeted the core functionality; but it was removed because it relied on a static database dump (which conformed to a specific schema) residing on the NERSC filesystem.

Related (bug report involving this endpoint): #723

**Comment by aclum** [comment]
I am advocating for this be escalated, this is the second time we've caught problem in production with this endpoint.  cc @shreddd

**Comment by eecavanna** [comment]
I'll create a branch that contains the skeleton of a test, and then work with @sujaypatil96 to flesh it out.

**Comment by eecavanna** [comment]
I created a branch (in a draft PR) in which I implemented the following two new tests:
1. `test_find_data_objects_for_study_having_none`: the `Study` exists, but doesn't have anything associated with it (e.g. no `Biosample`s, no `DataGeneration`s, no `DataObject`s). In this case, the endpoint returns no `DataObject`s.
2. `test_find_data_objects_for_study_having_one`: the `Study` exists, a `Biosample` is associated with that `Study`, a `MassSpectrometry` (which inherits from `DataObject`) has that `Biosample` as an input, and a `DataObject` is an output of that `MassSpectrometry`. In this case, the endpoint returns one `DataObject`.

Here's a link to the PR: https://github.com/microbiomedata/nmdc-runtime/pull/849/

I will invite others to contribute to it there.
